### ISSUE TITLE
Allow embedding nodes in other services / applications.

### ISSUE THREAD

**Issue by Zilog8** [slackhq/nebula#538]
In certain cases, it may not desirable or even possible to install a tun device to connect to a Nebula network.
A neat alternative would be to allow embedding of the Nebula service into other applications or services, which would then have individual access to resources on a Nebula network.
I have provided a pull-request with the functionality, and an example application that takes advantage of it.
  #537

**Comment by Zilog8** [comment]
I understand what you mean about the loss of the TCP stack. I've been trying to find a good replacement to plug into my application. Basically the following requirements:

1. "Reliable" transmission
2. Somewhat time-tested
2. Some pre-existing community momentum
3. Written in Go

I've narrowed down to [Pion's SCTP implementation](https://github.com/pion/sctp) , or Anacrolix's uTP implementations ([Pure Go](https://github.com/anacrolix/utp) / [CGO wrapper of libutp](https://github.com/anacrolix/go-libutp). They use the same interfaces, so supporting one means supporting the other). I started working with uTP, but life's been keeping me distracted lately and I haven't gotten anything functioning yet.
